### PR TITLE
Move `saf generate xccdf2inspec_stub` to use @mitre/inspec-objects

### PR DESCRIPTION
**Pull request by camdenmoors**
This improves CIS conversion and adds OVAL definition support
- [x] `toRuby` was updated, so needs to be reviewed to pass tests before merging
Need to make sure that all flags are manually tested
- [x] Test with -i only, xccdf file
- [x] Test help functions
- [ ] Test CIS 
- [x] Test Oval
- [x] Should we have the encodingHeader as an option? 
  - [x] I can add it underneath to the ts-inspec-objects function (Add issue for this)
  Yes, we should still have the ability to add a user defined header to files but we wouldn't add one by defaultl
- [x] Get rid of / address line length flag because it was unused
- [x] Update the ts-inspec-objects version after release of ts-inspec-objects (can do before or after)
  
Related to several issues:
#488

### CODE REVIEW

**Comment by aaronlippold** [comment]
Seems like we have some merge conflicts

**Comment by aaronlippold** [comment]
How come we got rid of this code? https://github.com/mitre/saf/pull/564/files#diff-9b4eacf4f12e389fa17ca7978d25358f3fa304f1b4ef7597ec398b0fbed998dcL51-L67

**Comment by em-c-rod** [comment]
> How come we got rid of this code? https://github.com/mitre/saf/pull/564/files#diff-9b4eacf4f12e389fa17ca7978d25358f3fa304f1b4ef7597ec398b0fbed998dcL51-L67

There were basically two different implementations of parsing and writing out the InSpec controls - one here and one in `ts-inspec-objects`. This PR replaces the implementation here with referencing that in `ts-inspec-objects` instead.

File: README.md [hunk 1]
```diff
@@ -1101,19 +1107,27 @@ EXAMPLES
 ```
 generate xccdf_benchmark2inspec_stub              Translate a DISA STIG XCCDF Benchmark XML file into a skeleton for an InSpec profile
 
-  USAGE
-    $ saf generate xccdf_benchmark2inspec_stub -i <stig-xccdf-xml> -o <output-folder> [-h] [-m <metadata-json>] [-s] [-r | -S] [-l <line-length>] [-e]
+USAGE
+  $ saf generate xccdf_benchmark2inspec_stub -i <stig-xccdf-xml> -o <output-folder> [-h] [-m <metadata-json>] [-T vuln|group|cis|stig] [-s] [-L info|warn|debug|verbose]
```
Comment (Amndeep7) [inline README.md:1111]: why are we changing the internal naming, but not the external function name?

**Review by Amndeep7 — CHANGES_REQUESTED** [review]

File: src/commands/generate/delta.ts [hunk 2]
```diff
@@ -10,17 +10,20 @@ export default class GenerateDelta extends Command {
 
   static flags = {
     help: Flags.help({char: 'h'}),
-    input: Flags.string({char: 'i', required: true, multiple: true, description: 'Input execution/profile JSON file(s) OR InSpec Profile Folder, and the updated XCCDF XML files'}),
+    input: Flags.string({char: 'i', required: true, multiple: true, description: 'Input execution/profile JSON file(s), InSpec Profile Folder, AND the updated XCCDF XML files'}),
```
[the remark this answers is not in the document]
Comment (em-c-rod) [reply src/commands/generate/delta.ts:13]: It has actually always been required, but this language was written incorrectly here. Fixing to be more clear.

File: README.md [hunk 3]
```diff
@@ -986,19 +986,25 @@ validate threshold            Validate the compliance and status counts of an HD
 ```
 Update an existing InSpec profile in-place with new XCCDF metadata
 
+USAGE
+  $ saf generate delta -i <value> [-h] [-r <value>] [-T vuln|group|cis|stig] [-L info|warn|debug|verbose]
```
[the remark this answers is not in the document]
Comment (aaronlippold) [reply README.md]: https://github.com/oclif/core/pull/188/files

[the remark this answers is not in the document]
Comment (aaronlippold) [reply README.md]: Looks like this has been added into oclif?

[the remark this answers is not in the document]
Comment (aaronlippold) [reply README.md]: Does this mean we get docopt standards by default

[the remark this answers is not in the document]
Comment (em-c-rod) [reply README.md]: This was resolved manually, but we added an issue here to update SAF CLI flags as a whole https://github.com/mitre/saf/issues/964

[the remark this answers is not in the document]
Comment (em-c-rod) [reply README.md]: Updated manually and added here for extra fixes https://github.com/mitre/saf/issues/964

File: src/commands/generate/xccdf_benchmark2inspec_stub.ts [hunk 4]
```diff
@@ -1,188 +1,129 @@
...
+        throw new Error('Passed OVAL definitions file does not exist')
       }
+    }
 
-      if (!extractedDescription.VulnDiscussion) {
-        throw new Error('Vulnerability exists without VulnDiscussion')
-      }
+    // Read the XCCDF file
+    const xccdf = fs.readFileSync(flags.input, 'utf8')
+    let profile
```
[the remark this answers is not in the document]
Comment (em-c-rod) [reply src/commands/generate/xccdf_benchmark2inspec_stub.ts]: Done

File: src/commands/generate/xccdf_benchmark2inspec_stub.ts [hunk 5]
```diff
@@ -1,188 +1,129 @@
...
-import YAML from 'yaml'
-import {CciNistMappingData} from '@mitre/hdf-converters'
+import {createWinstonLogger} from '../../utils/logging'
+import {processOVAL, processXCCDF} from '@mitre/inspec-objects'
 
-export default class XCCDFBenchmark2InSpec extends Command {
-  static usage = 'generate xccdf_benchmark2inspec_stub -i <stig-xccdf-xml> -o <output-folder> [-h] [-m <metadata-json>] [-s] [-r | -S] [-l <line-length>] [-e]'
+export default class XCCDF2InSpec extends Command {
+  static usage =
+    'generate xccdf_benchmark2inspec_stub -i, --input=XML -o, --output=FOLDER';
```
[the remark this answers is not in the document]
Comment (em-c-rod) [reply src/commands/generate/xccdf_benchmark2inspec_stub.ts]: Added an issue to resolve for all of SAF CLI https://github.com/mitre/saf/issues/964

File: README.md (hunk 1 above)
Comment (em-c-rod) [reply README.md]: Done

File: test/commands/generate/delta.test.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      path.resolve('./test/sample_data/xccdf/stigs/rhel-7-v3r8-mini-sample-xxcdf.xml'),
+      '-o',
+      `${tmpobj.name}`,
+      '-r',
+      `${tmpobj.name}`])
+    .it('should generate a report name delta.md and place it the default directory', () => {
+      expect(fs.lstatSync((`${tmpobj.name}/delta.md`)).isFile()).to.be.true // skipcq: JS-0354
+    })
+
+  // test
```
Comment (wdower) [inline test/commands/generate/delta.test.ts:70]: Seems like we've got a good handful of tests, but is there a reason this one is commented out and the rest are stubbed?

File: test/commands/generate/spreadsheet2inspec_stub.test.ts [hunk 7]
```diff
@@ -13,4 +13,12 @@ describe('Test spreadsheet2inspec_stub', () => {
       const fileCount = fs.readdirSync(`${tmpobj.name}/Ubuntu/controls/`).length
       expect(fileCount).to.eql(194)
     })
+
```
Comment (wdower) [inline test/commands/generate/spreadsheet2inspec_stub.test.ts:16]: Is this test still necessary? Can it be removed, or uncommented?

**Review by wdower — COMMENTED** [review]
LGTM, just wondering what's up with those commented out unit tests for `delta` and `spreadsheet2inspec_stub`.

**Pull request merged**